src/utils/mtev_hash.c

```diff
@@ -200,7 +273,16 @@ void mtev_hash_init_size(mtev_hash_table *h, int size) {
   mtevAssert(ck_hs_init(&h->hs, CK_HS_MODE_OBJECT | CK_HS_MODE_SPMC, hs_hash, hs_compare, &my_allocator,
                          size, lrand48()));
   mtevAssert(h->hs.hf != NULL);
+
+  mtev_hash_set_lock_mode_funcs(h, lock_mode);
+}
+
+void
+mtev_hash_set_lock_mode(mtev_hash_table *h, mtev_hash_lock_mode_t lock_mode)
+{
+  mtev_hash_set_lock_mode_funcs(h, lock_mode);
```

Comment:
should probably call `destroy_locks` in this function, before calling `set_lock_mode_funcs`? or else `set_lock_mode_funcs` should call `destroy_locks` internally?